packages/alfa-dom/test/node.spec.tsx

```diff
@@ -167,3 +168,35 @@ test(`Node.clone() correctly replaces elements based on predicate`, (t) => {
     children: [...newElements.map((e) => e.toJSON()), bar.toJSON()],
   });
 });
+
+test(`#toJSON() serializes boxes of all descendants when device is passed in`, (t) => {
```

Comment:
**Suggestion:** It seems that DOM children were already correctly handled 🤔 (was this example failing?) This PR is fixing the recursion for Shadow DOM and content document, we should add tests for these instead (also with `<slot>`).

Reply:
1. I don't think they were. I think passing the `options` to `...super.toJSON(option)` on line 302 in element.ts is was what causes the option to be passed to the children. When I remove it again it also fails this test.
2. I agree there should also be tests for shadow DOM and content and `<slot>`